src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -27,8 +27,8 @@ std::shared_ptr<ov::Node> change_constant_precision_to_fp16(std::shared_ptr<ov::
 
     int num_out_of_range = 0;
     for (size_t i = 0; i < size; ++i) {
-        // if it is smaller than the smallest positive normal fp16 but not zero
-        if (std::abs(src_data[i]) <= ov::float16::from_bits(0x0400) && src_data[i] != 0.0f) {
+        // if abs value is smaller than the smallest positive fp16, but not zero
+        if (std::abs(src_data[i]) < ov::float16::from_bits(0x0001) && src_data[i] != 0.0f) {
```

Comment:
I see the different meaning about the minimum positive FP16 number. Please see: https://stackoverflow.com/questions/56711084/represents-fp16-minimum-number-in-hex-format
Your current version is aligned with it.